crates/core/plugin_sm/src/operation_logs.rs

```diff
@@ -1,29 +1,32 @@
-use plugin_sm::log_file::LogFile;
-use std::cmp::Reverse;
 use std::collections::BinaryHeap;
 use std::path::PathBuf;
+use std::{cmp::Reverse, collections::HashMap};
 use time::{format_description, OffsetDateTime};
 use tracing::log;
 
+use crate::log_file::LogFile;
+
 #[derive(Debug, thiserror::Error)]
 pub enum OperationLogsError {
     #[error(transparent)]
     FromIo(#[from] std::io::Error),
 
     #[error(transparent)]
     FromTimeFormat(#[from] time::error::Format),
+
+    #[error("Operation name is None. Expected Some(`value`).")]
+    NoOperation,
 }
 
 #[derive(Debug)]
 pub struct OperationLogs {
-    log_dir: PathBuf,
-    max_update_logs: usize,
-    max_list_logs: usize,
+    pub log_dir: PathBuf,
 }
 
 pub enum LogKind {
     SoftwareUpdate,
     SoftwareList,
+    Operation,
 }
```

Comment:
See my comment above.

```suggestion
pub enum LogKind {
    SoftwareUpdate,
    SoftwareList,
    Operation(String),
}
```